core/include/detray/geometry/detector_volume.hpp

```diff
@@ -15,6 +15,49 @@
 
 namespace detray {
 
+namespace detail {
+
```

Comment:
Split the implementation of visiting all surfaces in the volume (i.e. querying all the surface finders of the volume for all surfaces) from the neighborhood lookup (i.e. querying all the surface finders of the volume for a neighborhood) from the implementation of the surface store tuple unrolling, so that it can be reused for both cases. The ```functor``` that is passed is a customization point. The neighborhood lookup is only called with the default ```candidate_search``` functor of the navigator so far.